core-bundle/src/Resources/contao/library/Contao/Input.php

```diff
@@ -56,9 +56,52 @@ class Input
 	 */
 	public static function initialize()
 	{
-		$_GET    = static::cleanKey($_GET);
-		$_POST   = static::cleanKey($_POST);
-		$_COOKIE = static::cleanKey($_COOKIE);
+		$_GET    = static::cleanKeyInternal($_GET);
+		$_POST   = static::cleanKeyInternal($_POST);
+		$_COOKIE = static::cleanKeyInternal($_COOKIE);
+	}
+
+	public static function encodeInput(array|string $value, InputEncodingMode $mode, bool $encodeInsertTags = true): array|string
```

Comment:
Should we maybe simplify this signature to `encodeInput(string $value, …): string` for the sake of type static safety?

IMHO any code using arrays can still do `array_map(fn($v) => Input::encodeInput($v), $values)` which is a reasonable overhead.

Reply:
Done in 9ffc6fdb963ad41d73cf6a19bd36e21ae15ec11b